pynxtools/testing/nexus_conversion.py

```diff
@@ -94,15 +95,15 @@ def convert_to_nexus(self, ignore_undocumented=False):
             template=Template(template), file_paths=tuple(input_files)
         )
 
+        entry_names = read_data.get_all_entry_names()  # type: ignore[attr-defined]
```

Comment:
@RubelMozumder I added this here because this is [part of the regular nexus converter](https://github.com/FAIRmat-NFDI/pynxtools/blob/c8e3c8b45e3cb5a738ea54b38a821cd62a667916/pynxtools/dataconverter/convert.py#L169) (so that every nexus file file gets the version by default) and thus it makes sense to add this also during the testing.

Reply:
I would say
to replace lines 92-100 add the code 
```python
    data = transfer_data_into_template(
        input_file=input_file,
        reader=reader,
        nxdl_name=nxdl,
        nxdl_root=nxdl_root,
        skip_verify=skip_verify,
        **kwargs,
    )
``` 

 from https://github.com/FAIRmat-NFDI/pynxtools/blob/c8e3c8b45e3cb5a738ea54b38a821cd62a667916/pynxtools/dataconverter/convert.py#L221C1-L228C6. Because, we should not right the same code different place, but need to reuse them . The benefit is in the future we can change in a single place and automatically reused everywhere.

Reply:
I agree, this makes the most sense. I have implemented it here. In that case, the `reader ` passed to the ReaderTest object must however not be the class, but the name of the reader. This is because within `transfer_data_into_template`, the `get_reader` function is called. See my changes to the documentation how this influences the implementation.